packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -428,7 +428,7 @@ class RenderEditable extends RenderBox {
   int _handleHorizontalArrows(bool rightArrow, bool leftArrow, bool shift, int newOffset) {
     // Set the new offset to be +/- 1 depending on which arrow is pressed
     // If shift is down, we also want to update the previous cursor location
-    if (rightArrow && _extentOffset < text.text.length) {
+    if (rightArrow && _extentOffset < text.toPlainText().length) {
```

Comment:
If the textSpan has children instead of text, we need to use this function to get the string.
This can happen when textEditingValue has composing range.

Reply:
Sounds good. It looks like `toPlainText` is how we get the length in the rest of the file.